prov/efa/test/efa_unit_test_ep.c

```diff
@@ -12,19 +12,38 @@
  */
 void test_rxr_ep_host_id(struct efa_resource **state, bool file_exists, char *raw_id, uint64_t expect_id)
 {
+	int fd;
+	ssize_t written_len;
+	const size_t host_id_file_length = 10;
+	char *host_id_file;
 	struct efa_resource *resource = *state;
 	struct rxr_ep *rxr_ep;
-	size_t file_name_length = 10;
-	char host_id_file[file_name_length];
-	FILE *f_handle;
 
 	rxr_env.host_id_file = NULL;
 
 	if (file_exists) {
-		new_temp_file(host_id_file, file_name_length);
-		f_handle = fopen(host_id_file, "w");
-		fprintf(f_handle, raw_id);
-		fclose(f_handle);
+		host_id_file = malloc(host_id_file_length + 1);
```

Comment:
Why not just use a string  like

```
char host_id_file[] = "XXXXXXXXX"
```
?

Reply:
> Why not just use a string like

Thanks! I tried `char *host_id_file = "xxxxx"` and it does not work for `mkstemp`, but `host_id_file[]` does!

Will update.

Reply:
@wenduwan @wzamazon 

> char host_id_file[] = "XXXXXXXXX"

My understanding is that mkstemp would modify such a string in-place, and that `host_id_file[]` and `char *host_id_file` are both equivalent in that they simply point to a portion of the binary which has the string "XXXXXXXXXX" sitting.  I would then expect that the first call to mkstemp would succeed in modifying the string to "XXXX7d8c7s" but then subsequent calls would fail because the (persistent) string doesn't end in 6 "X" characters.

Maybe `char host_id_file[]` is actually setting some local memory each time??  I'm surprised this PR works as is.

Reply:
So far the tests have passed(with multiple calls to mkstemp) so I think this issue did not happen.

My understanding is `char host_id_file[]`(not static) is reallocated in each test call, so it is always initialized to `XXXXXXXXXX`